packages/devtools_server/lib/src/external_handlers.dart

```diff
@@ -25,36 +27,73 @@ import 'usage.dart';
 ///
 /// This serves files out from the build results of running a pub build of the
 /// DevTools project.
-Future<shelf.Handler> defaultHandler(ClientManager clients) async {
+Future<shelf.Handler> defaultHandler(
+  ClientManager clients, {
+  bool debugMode = false,
+}) async {
   final resourceUri = await Isolate.resolvePackageUri(
       Uri(scheme: 'package', path: 'devtools/devtools.dart'));
 
   final packageDir = path.dirname(path.dirname(resourceUri.toFilePath()));
 
   // Default static handler for all non-package requests.
-  final buildDir = path.join(packageDir, 'build');
-  final buildHandler = createStaticHandler(
-    buildDir,
-    defaultDocument: 'index.html',
-  );
+  Handler buildDirHandler;
+  if (!debugMode) {
+    buildDirHandler = createStaticHandler(
+      path.join(packageDir, 'build'),
+      defaultDocument: 'index.html',
+    );
+  }
+
+  Handler debugProxyHandler;
+  if (debugMode) {
+    // Start up a flutter run -d web-server instance.
+
+    // ignore: unawaited_futures
+    Process.start(
+      'flutter',
+      ['run', '-d', 'web-server', '--web-port=9101'],
+      workingDirectory: path.join('..', 'devtools_app'),
+    ).then((Process process) {
+      // Write all flutter run process output to the server's output.
+      process.stdout.transform(utf8.decoder).listen(stdout.write);
+      process.stderr.transform(utf8.decoder).listen(stderr.write);
+
+      // Proxy all stdin to the flutter run process's input.
+      //stdin.pipe(process.stdin);
+      stdin.lineMode = false;
+      stdin.listen((event) {
+        process.stdin.add(event);
+      });
+
+      // Exit when the flutter run process exits.
+      process.exitCode.then(exit);
+    });
+
+    debugProxyHandler = proxyHandler(Uri.parse('http://localhost:9101/'));
```

Comment:
pull 9101 out into a constant and then inject it using string interpolation here and above. That way if we ever change the port number we make sure we change it in both places.

Reply:
sounds good - moved to using a const value